perfkitbenchmarker/benchmarks/cassandra_ycsb_benchmark.py

```diff
@@ -0,0 +1,187 @@
+  """Creates a Cassandra table for use with YCSB."""
+  template_path = data.ResourcePath(CREATE_TABLE_SCRIPT)
+  remote_path = os.path.join(
+      cassandra.CASSANDRA_DIR,
+      os.path.basename(os.path.splitext(template_path)[0]))
+  vm.RenderTemplate(template_path, remote_path,
+                    context={'keyspace': keyspace,
+                             'column_family': column_family,
+                             'replication_factor': replication_factor})
+
+  cassandra_cli = os.path.join(cassandra.CASSANDRA_DIR, 'bin', 'cassandra-cli')
+  command = '{0} -f {1} -h {2}'.format(cassandra_cli, remote_path,
+                                       vm.internal_ip)
+  vm.RemoteCommand(command, should_log=True)
+
+
+def _GetVMsByRole(vms):
+  """Gets a dictionary mapping role to a list of VMs."""
+  cassandra_vms = vms[:-FLAGS.ycsb_client_vms]
+  return {'vms': vms,
+          'cassandra_vms': cassandra_vms,
+          'seed_vm': cassandra_vms[0],
+          'non_seed_cassandra_vms': cassandra_vms[1:],
+          'loaders': vms[-FLAGS.ycsb_client_vms:]}
+
+
+def Prepare(benchmark_spec):
+  """Prepare the virtual machines to run YCSB against Cassandra.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+  """
+  vms = benchmark_spec.vms
+  by_role = _GetVMsByRole(benchmark_spec.vms)
+
+  loaders = by_role['loaders']
+  assert loaders, vms
+
+  # Cassandra cluster
+  cassandra_vms = by_role['cassandra_vms']
+  assert cassandra_vms, 'No Cassandra VMs: {0}'.format(by_role)
+  seed_vm = by_role['seed_vm']
+  assert seed_vm, 'No seed VM: {0}'.format(by_role)
+
+  install = functools.partial(_InstallCassandra, seed_vms=[seed_vm])
+  vm_util.RunThreaded(install, cassandra_vms)
+
+  cassandra.StartCluster(seed_vm, by_role['non_seed_cassandra_vms'])
+
+  _CreateYCSBTable(seed_vm)
+
+  vm_util.RunThreaded(_InstallYCSB, loaders)
+
+
+def Run(benchmark_spec):
+  """Spawn YCSB and gather the results.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+
+  Returns:
+    A list of sample.Sample instances.
+  """
+  vms = benchmark_spec.vms
+  loaders = _GetVMsByRole(vms)['loaders']
+  cassandra_vms = _GetVMsByRole(vms)['cassandra_vms']
+  logging.debug('Loaders: %s', loaders)
+  vms = benchmark_spec.vms
+
+  executor = ycsb.YCSBExecutor(
+      'cassandra-10',
+      hosts=','.join(vm.internal_ip for vm in cassandra_vms))
+
+  kwargs = {'hosts': ','.join(vm.internal_ip for vm in cassandra_vms),
+            'columnfamily': COLUMN_FAMILY,
+            'cassandra.readconsistencylevel': READ_CONSISTENCY,
+            'cassandra.scanconsistencylevel': READ_CONSISTENCY,
+            'cassandra.writeconsistencylevel': WRITE_CONSISTENCY,
+            'cassandra.deleteconsistencylevel': WRITE_CONSISTENCY}
+
+  metadata = {'ycsb_client_vms': FLAGS.ycsb_client_vms,
+              'num_vms': FLAGS.num_vms,
```

Comment:
Above, you use num_vms = max(FLAGS.num_vms, 3) . Have you confirmed that the metadata contains the actual number used if FLAGS.num_vms uses a default less than three? Would be confusing otherwise.